src/validation.h

```diff
@@ -994,6 +994,16 @@ class ChainstateManager
      */
     bool ProcessNewBlockHeaders(const std::vector<CBlockHeader>& block, BlockValidationState& state, const CChainParams& chainparams, const CBlockIndex** ppindex = nullptr) LOCKS_EXCLUDED(cs_main);
 
+    /**
+     * Try to add a transaction to the memory pool.
+     *
+     * @param[in]  tx              The transaction to submit for mempool acceptance.
+     * @param[in]  bypass_limits   When true, don't enforce mempool fee and capacity limits.
+     * @param[in]  test_accept     When true, run validation checks but don't submit to mempool.
+     */
+    [[nodiscard]] MempoolAcceptResult ProcessTransaction(const CTransactionRef& tx, bool bypass_limits, bool test_accept=false)
```

Comment:
In commit "[validation] Add CChainState::ProcessTransaction()" (bbc785f31b5d70c702ae37d259c46652e243ddba)

Maybe for consistency with `ProcessNewBlock` and `ProcessNewBlockHeaders`, call this method `ProcessNewTransaction` instead of `ProcessTransaction`. Might also help suggest that purpose of processing is to add new data, not validate existing data.

Reply:
Haha you noticed. I actually hope that we'd go the other way and change `ProcessNewBlock` to `ProcessBlock` and `ProcessNewBlockHeaders` to `ProcessBlockHeaders`. There's nothing to prevent passing in the same block/headers/transaction that have previously been processed, and the functions have ways of returning that information to the caller (`ProcessNewBlock` will set the `new_block` out-param to false and `ProcessTransaction` will set the `TxValidationState` result to `TX_CONFLICT` to indicate "Tx already in mempool or conflicts with a tx in the chain"). I think the `New` in `ProcessNew` is therefore redundant and perhaps even a bit misleading.